Rework comparison with other frameworks (#1075)
grammyjs/website: site/docs/resources/comparison.md

- grammY integrates hints from the [official Bot API reference](https://core.telegram.org/bots/api) inline that help you while you're coding.
  Telegraf does not give you any explanations on your code.
- Many more things like better performance, a large plugin ecosystem, documentation that is translated for billions of people, better integration with databases and web frameworks, better runtime compatibility, a [VS Code extension](https://marketplace.visualstudio.com/items?itemName=grammyjs.grammyjs), and a number of other things that you will discover as you go.

Here is a list of reasons why you should use Telegraf instead of grammY.

- You already have a large bot written in Telegraf and you no longer really work on it.
  In that case, migrating to grammY may take more time than you will save in the long run, no matter how smooth the migration is.
- You know Telegraf like the back of your hand and you do not care about changing your skill set.
  grammY introduces a number of novel concepts that can be unfamiliar if you have used Telegraf only, and using grammY means that you will be exposed to new things.

> - There are a few details where Telegraf and grammY use different syntax to achieve the same thing, and you just happen to prefer one style of the other.

Reviewer: ```suggestion
- There are a few details where Telegraf and grammY use different syntax to achieve the same thing, and you just happen to prefer one style over the other.
```

### Other Frameworks

There currently are no other TypeScript libraries that are worth using for building bots.
Everything except grammY, Telegraf, and NTBA is largely unmaintained and thus horribly out of date.

Did you just create a new awesome library and we are not aware of it yet?
Feel free to edit this page and add a comparison---or tell us what you think in the [group chat](https://t.me/grammyjs)!

## Comparison With Frameworks in Other Programming Languages

There are reasons to favor a different programming language over TypeScript.
The most important thing is that you like working with your tools and languages.
If you are determined to stick with a different language, then you can stop reading here.

Given that you're still reading, you may want to know why grammY is written in TypeScript, and why you should maybe consider picking this language for your bot, too.

This section will outline how TypeScript has a few advantages over other languages when it comes to developing Telegram bots.
This comparison will be limited to Python, Go, and Rust.
Feel free to add more sections if you want to contrast TypeScript with another language.

> Some of the following points are based partially based on personal opinions.

Reviewer: ```suggestion
Some of the following points are partially based on personal opinions.
```

   Right now, V8 and its competitors make JavaScript the world's fastest scripting language.
   If you like your bot to be as fast as possible while still enjoying a dynamic language, then grammY is your best bet.

As always, programming languages excel at certain tasks and should be avoided for others.
This is no exception.

For example, with the current state of the ecosystems, anything related to machine learning should not be done in JavaScript.
However, when it comes to web servers, TypeScript tends to be a much better choice.

### Frameworks Written in Go

If you are proficient in both TypeScript and Go, then a reasonable metric for deciding on a language for your bot is the balance between development speed and execution speed.

Pick grammY if you are not completely sure what you are building.
TypeScript lets you iterate on your code base at incredible speeds.
It is great for rapid prototyping, trying out new things, getting to know bots, and getting things done quickly.
As a rule of thumb, processing ~100,000,000 updates per day can be done easily with TypeScript, but going beyond that will require extra work, such as using one more grammY plugin.

> Pick a library written in Go if you already know fairly well what you will be building (you don't expect to need much assistance), and you alreay know that your bot will process very high numbers of updates.

Reviewer 2: ```suggestion
Pick a library written in Go if you already know fairly well what you will be building (you don't expect to need much assistance), and you already know that your bot will process very high numbers of updates.
```

> Pick a library written in Go if you already know fairly well what you will be building (you don't expect to need much assistance), and you already know that your bot will process very high numbers of updates.

Author: ```suggestion
Pick a library written in Go if you already know fairly well what you will be building (you don't expect to need much assistance), and you already know that your bot will process a very large number of updates.
```

Author: @roj1512 accept this if you think that it is an improvement to the wording

As a natively compiled language, Go outperforms TypeScript at raw CPU speed by several orders of magnitude.
This is much less relevant when you write a bot because most of the time is spent waiting for the network, but eventually, it will start to matter how fast your bot can parse JSON.
Go can be a better choice in these cases.

### Frameworks Written in Rust

A similar point can be made [as with Go](#frameworks-written-in-go), but it is even stronger with Rust.
In a way, it will take you even more time to write Rust, but your bot will be even faster, too.

Also, please not that using Rust is fun but rarely necessary for bots.

> If you want to use Rust, then do it, but consider saying that you love Rust and not that it is the right tool for the job.

Reviewer 3: `please not that`?

Is this correct?

Author: No. It should be _please note that_

Reviewer 3: ```suggestion
Also, please note that using Rust is fun but rarely necessary for bots.
If you want to use Rust, then do it, but consider saying that you love Rust and note that it is the right tool for the job.
```

Author: Just here: 
```suggestion
Also, please note that using Rust is fun but rarely necessary for bots.
If you want to use Rust, then do it, but consider saying that you love Rust and not that it is the right tool for the job.
```
